jax/interpreters/pxla.py

```diff
@@ -345,30 +336,56 @@ def block_until_ready(self):
   @property
   def _value(self):
     if self._npy_value is None:
-      ids = self._ids()
+      # TODO(skye): remove this to avoid transferring replicated buffers?
       self.copy_to_host_async()
-      self._npy_value = self._collect([self.device_buffers[i].to_py() for i in ids])
+      self._npy_value = onp.empty(self.aval.shape, self.aval.dtype)
+      # TODO(skye): benchmark and possibly switch to a set (maybe with with
+      # cached hashable indices?)
+      already_copied_indices = []
+      for buf, idx in zip(self.device_buffers, self.logical_indices):
+        if idx in already_copied_indices:
+          continue
```

Comment:
This will only have effect if the same exact index is used on another buffer. When there are overlapping indices it will copy and over write multiple times. Is it guaranteed that the overlapping buffers have the same data?